### PR TITLE
[Documentation] Concepts: Coupons

### PR DESCRIPTION
**Pull request by TheMadeleine**
| Q               | A
| --------------- | ---
| Doc fix?        | no
| New docs?    | yes
| BC breaks?      | no
| Related tickets | partially #5275 
| License         | MIT

### CODE REVIEW

File: docs/book/coupons.rst [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+Coupon Parameters
+-----------------
+
+A **Coupon** besides a ``code`` has a date when it expires, the ``usageLimit`` and it counts how many times it was already used.
+
+How to create a Coupon programmatically?
+----------------------------------------
+
+Just as usually, use a factory. The coupon needs a ``code`` and a ``name``.
```
Comment (pjedrzejewski) [inline docs/book/coupons.rst]: Does not need ``name``. :)

File: docs/book/coupons.rst [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+A **Coupon** besides a ``code`` has a date when it expires, the ``usageLimit`` and it counts how many times it was already used.
+
+How to create a Coupon with a Promotion programmatically?
+---------------------------------------------------------
+
+.. warning::
+
+   The Promotion has to be ``couponBased = true`` in order to be able to hold a collection of Coupons that belong to it.
+
+Let's create a promotion that will have one coupon that activates the free shipping.
```
Comment (NeverResponse) [inline docs/book/coupons.rst]: How 'bout `Let's create a promotion with a single/one coupon that/which...`?

File: docs/book/coupons.rst [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+   $actionFactory = $this->container->get('sylius.factory.promotion_action');
+
+   // Provide the amount in float ( 1 = 100% )
+   $action = $actionFactory->createPercentageShippingDiscount(1);
+
+   $promotion->addAction($action);
+
+   $this->container->get('sylius.repository.promotion')->add($promotion);
+
+Finally to see the effects of your promotion with Coupon you need to **apply a coupon on the Order**.
```
Comment (NeverResponse) [inline docs/book/coupons.rst]: Why is the coupon capitalized, but promotion not?

File: docs/book/coupons.rst [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+.. code-block:: php
+
+   $order->setPromotionCoupon($coupon);
+
+   $this->container->get('sylius.order_processing.order_processor')->process($order);
+
+Coupon Generator
+----------------
+
+Making up new codes might become difficult if you would like to generate a lot of coupons at once. That is why Sylius
```
Comment (NeverResponse) [inline docs/book/coupons.rst]: `...if you would like to create/make a lot...`
Sentence `if you would like to generate... use the generator`l, might hit on readers' intelligence in a `no shit sherlock` way

**Comment by pjedrzejewski** [comment]
Nice work Magda, thanks!

**Pull request merged**